blaze/server/client.py

```diff
@@ -100,16 +108,18 @@ def compute_down(expr, ec, **kwargs):
     from .server import to_tree
     tree = to_tree(expr)
 
-    r = requests.get('%s/compute.json' % ec.url,
-                     data=json.dumps({'expr': tree}),
-                     headers={'Content-Type': 'application/json'})
+    serial = ec.serial
+    r = requests.get('%s/compute.%s' % (ec.url, ec.mode),
+                     data=serial.dumps({'expr': tree}))
```

Comment:
Note that for pickle, if the client and server are close and if you have a decent amount of numerical data, and if you're in Python2, then you *really* want to use the following:

    dumps = partial(pickle.dumps, protocol=pickle.HIGHEST_PROTOCOL)